webapp/src/components/confirm/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { ReactNode, useEffect, useRef, useState } from "react";
+import ReactDOM, { Root } from "react-dom/client";
+
+import { Button } from "react-bootstrap";
+
+import "./confirm.css";
+
+export type ConfirmProps = {
+  title: string|ReactNode,
+  message?: string|ReactNode,
+  confirmButton: {
+    text: string|ReactNode,
+    action: () => void
+  },
+  cancelButton: {
+    text: string|ReactNode,
+    action?: () => void
+  },
+  onShow?: () => void
+};
+function Confirm({ title, message, confirmButton, cancelButton, onShow }: ConfirmProps) {
+  const [visible, setVisible] = useState(false);
+  
+  const overlay = useRef<HTMLDivElement|null>(null);
+  
+  const close = () => {
+    setVisible(false);
+    setTimeout(removeConfirmElement, 250);
+  }
+  
+  useEffect(() => {
+    setVisible(true);
+    if(onShow) onShow();
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div
+      className="confirm"
+      ref={overlay}
+      onClick={(event) => {
+        if (event.target === overlay.current) {
+          close();
+        }
+      }}
+    >
+      <div
+        className="confirmCard"
+        style={visible ? {opacity: 1} : {opacity: 0}}
+      >
+        <div className="confirmCardContent">
```

Comment:
This looks great, and we definitely can reuse this!

The output of this card is close to react bootstrap's card, do you think we would be able to use that instead?  If not, some ways I think the styling would be improved:

* Less heavy font weight for the header
* More padding around the card body
* Larger buttons, maybe justified with space-around?

Completely optional

Reply:
I Can replicate the styling as close as possible to bootstrap's styling, otherwise would need to reshuffle the divs/elements around to make it work with bootstrap itself since this is an dynamic element which only attaches and deattaches to the app body.
Styling replication would be simpler i think, making it use bootstraps elements would be a lil involved but still doable, lemme know which one you would prefer! :)